docs/route-tasks.md

```diff
@@ -0,0 +1,171 @@
+WorkerFactory factory = WorkerFactory.newInstance(client);
+// Create a Worker that is subscribed to a shared Task Queue
+final Worker workerForCommonTaskList = factory.newWorker(globalTaskQueue);
+// Register Workflow and Activities
+// Create a Worker that is subscribed to a private TaskQueue
+final Worker workerForHostSpecificTaskList = factory.newWorker(hostSpecifiTaskQueue);
+// Register activities
+```
+
+In the [Workflow](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/FileProcessingWorkflowImpl.java#L42), the first Activity Task is executed by the Worker listening to the global Task Queue:
+
+```java
+public FileProcessingWorkflowImpl() {
+    // Create activity clients.
+    ActivityOptions ao =
+        ActivityOptions.newBuilder()
+        // Send the Activity to the shared Task Queue
+        .setTaskQueue(globalTaskQueue)
+        .build();
+    this.defaultTaskQueueStore = Workflow.newActivityStub(StoreActivities.class, ao);
+}
+```
+
+The Activities should be designed to [return an object](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/StoreActivities.java) that includes both the Activity result and the name of the host-specific Task Queue to which the next Activity Task will be sent.
+
+```java
+public interface StoreActivities {
+
+  final class ResultAndTaskQueuePair {
+    // Name of the host-specific  Task Qeueu
+    private final String hostspecificTaskQueue;
+    // This is the Activity result (e.g. the name or location of the file)
+    private final String activityResult;
+
+    // Get and set methods ...
+  }
+
+  // Interface methods ....
+}
+```
+
+The host-specific Task Queue is then set on [subsequent Activities](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/FileProcessingWorkflowImpl.java#L63):
+
+```java
+private void processFileImpl(URL source, URL destination) {
+    // Initialize stubs that are specific to the returned task queue.
+    ActivityOptions hostActivityOptions =
+        ActivityOptions.newBuilder()
+        // Set the host-specific Task Queue
+        .setTaskQueue(hostSpecificTaskQueue)
+        .build();
+
+    // Call the Activity to process the file using worker-specific task queue.
+}
+```
+
+### Go
+
+In Go, you have two options.
+
+**Option 1**: Implement the same design as you would in Java, which is to create two Workers (one listens to the shared Task Qeue and the other listens to the private Task Queue) and return the name of the private Task Queue back to the Workflow as an Activity result.
+
+**Option 2**: Create a Workflow Session which automatically ties any remaining Workflow Activity executions to the Worker that started the Workflow. The [FileProcessing sample in Go](https://github.com/temporalio/temporal-go-samples/tree/master/fileprocessing) is a good example of sessions being used for this purpose.
```

Comment:
"any remaining Workflow Activity executions "

I think "remaining" is misleading